### PR TITLE
[build] Always build result package dependencies

### PR DESCRIPTION
**Pull request by pjcollins**
Sometimes 'xaprepare.exe' will fail before all of the build status
packaging dependencies are built. We can fix some of these cases by
ensuring that we always call the build target before calling any status
packaging targets.

### CODE REVIEW

File: build-tools/scripts/PrepareWindows.targets [hunk 1]
```diff
@@ -5,7 +5,9 @@
     <_TopDir>$(MSBuildThisFileDirectory)..\..</_TopDir>
     <_XAPrepareExe>$(MSBuildThisFileDirectory)..\xaprepare\xaprepare\bin\$(Configuration)\xaprepare.exe</_XAPrepareExe>
     <_XAPrepareStandardArgs>--no-emoji --run-mode=CI -a</_XAPrepareStandardArgs>
```
Comment (jonathanpeppers) [inline build-tools/scripts/PrepareWindows.targets]: @grendello I think you mentioned we could put `-a` only if running on CI, is that correct?

If so, we should fix that here.

Comment (pjcollins) [reply build-tools/scripts/PrepareWindows.targets]: I've moved all of the params behind the new CI flag.

**Comment by jonpryor** [comment]
**Xamarin.Android (Windows Build and Test)** failures look like they should be addressed by PR #3262.

**Comment by pjcollins** [comment]
@jonpryor I think this is good to go, Windows is still green.

**Pull request merged**